checks/https.go

```diff
@@ -0,0 +1,271 @@
+	if err != nil || len(addrs) < 1 {
+		msg := "failed to resolve hostname to IP."
+		p.Result.Error = &msg
+		return nil
+	}
+	dnsResolve := time.Since(step).Seconds() * 1000
+	p.Result.DNS = &dnsResolve
+	
+	validate, _ := strconv.ParseBool(p.ValidateCert)
+	tlsConfig := &tls.Config{
+		InsecureSkipVerify: !validate,
+		ServerName: p.Host,
+	}
+	
+	// Dialing
+	start := time.Now()
+	conn, err := tls.Dial("tcp", addrs[0]+":"+p.Port, tlsConfig)
+	if err != nil {
+		msg := err.Error()
+		p.Result.Error = &msg
+		return nil
+	}
+	defer conn.Close()
+	connecting := time.Since(start).Seconds() * 1000
+	p.Result.Connect = &connecting
+
+	// Send
+	step = time.Now()
+	if err := request.Write(conn); err != nil {
+		msg := err.Error()
+		p.Result.Error = &msg
+		return nil
+	}
+	send := time.Since(step).Seconds() * 1000
+	p.Result.Send = &send
+	
+	// Wait & Receive
+	step = time.Now()
+	
+	// read first byte data
+	firstData := make([]byte, 1)
+	_, err = conn.Read(firstData)
+	if err != nil {
+		if err != io.EOF {
+			msg := "Read error"
+			p.Result.Error = &msg
+			return nil
+		}
+	}
+	wait := time.Since(step).Seconds() * 1000
+	p.Result.Wait = &wait
+
+	var buf bytes.Buffer
+	buf.Write(firstData)
+	data := make([]byte, 1024)
+	limit := 100 * 1024
+	for {
+		n, err := conn.Read(data)
+		if err != nil {
+			if err != io.EOF {
+				msg := "Read error"
+				p.Result.Error = &msg
+				return nil
+			}else{
+				break
+			}
+		}
+		buf.Write(data[:n])
+		if buf.Len() > limit {
+			conn.Close()
+			break
+		}
+	}
+	
+	recv := time.Since(step).Seconds() * 1000
+	/* 
+		Total time 
+	*/
+	total := time.Since(start).Seconds() * 1000
+	p.Result.Total = &total
+	
+	p.Result.Recv = &recv
+	
+	readbuffer := bytes.NewBuffer(buf.Bytes())
+	response, err := http.ReadResponse(bufio.NewReader(readbuffer), request)
+	
+	if err != nil {
+		msg := err.Error()
+		p.Result.Error = &msg
+		return nil
+	}
+	
+	var reader io.ReadCloser
+	if p.ExpectRegex != "" {
+		// Handle gzip
+		switch response.Header.Get("Content-Encoding") {
+			case "gzip":
+				reader, err = gzip.NewReader(response.Body)
+				defer reader.Close()
+				if err != nil {
+					msg := err.Error()
+					p.Result.Error = &msg
+					return nil
+				}
+			default:
+				reader = response.Body
+		}
+	}else{
+		reader = response.Body
+	}
+	
+	defer response.Body.Close()
+	body, err := ioutil.ReadAll(reader)
+	if err != nil {
+		msg := err.Error()
+		p.Result.Error = &msg
+		return nil
+	}
+	
+	// Data Length
+	dataLength := float64(0)
+	if dataLength, err = strconv.ParseFloat(response.Header.Get("Content-Length"), 64); dataLength < 1 || err != nil {
+		dataLength = float64(len(body))
+	}
+    p.Result.DataLength = &dataLength
+	
+	// Error response
+	statusCode := float64(response.StatusCode)
+	p.Result.StatusCode = &statusCode
+	if statusCode >= 400 {
+		msg := "Invalid status code " + strconv.Itoa(response.StatusCode);
+		p.Result.Error = &msg
+		return nil
+	}
+	
+	if response.TLS == nil {
```

Comment:
this wont work.  The response object was manually created by passing in the bytes read from the connection, it has no knowledge of the connection properties, including TLS.

You will need to get the PeerCertificates from the "conn" object created by `tls.Dial`